### ISSUE TITLE
how can i set fad in fad  out effect on every image , do i have to change page fragment??

### ISSUE THREAD

**Issue by ghost** [Cleveroad/slidingtutorial-android#31]

**Comment by Iojjj** [comment]
Hi @dixitsurabhi,

you can't do it in current version of library (1.0.3). We're setting our own [PageTransformer](https://developer.android.com/reference/android/support/v4/view/ViewPager.PageTransformer.html) to `ViewPager` so there is not way to wrap our or yours `PageTransformer` and use both of them at the moment. We'll look into this and add a way to specify custom `PageTransformer`.

What you can do for now: check [the sources of ViewPager](https://github.com/android/platform_frameworks_support/blob/master/v4/java/android/support/v4/view/ViewPager.java#L1890-L1901). Here you can see how they call a `PagerTransformer`, and you can implement fade animation the same way. For example, in your implementation of `TutorialFragment` or `TutorialSupportFragment`:

``` JAVA
@Override
public View onViewCreated(View view, Bundle savedInstanceState) {
    final viewPager = getViewPager();
    viewPager.addOnPageChangeListener(new ViewPager.SimpleOnPageChangeListener() {
        @Override
        public void onPageScrolled (int position, float positionOffset, int positionOffsetPixels) {
            final int scrollX = viewPager.getScrollX();
            final int childCount = viewPager.getChildCount();
            for (int i = 0; i < childCount; i++) {
                final View child = viewPager.getChildAt(i);
                final ViewPager.LayoutParams lp = (ViewPager.LayoutParams) child.getLayoutParams();
                if (lp.isDecor) continue;
                final float transformPos = (float) (child.getLeft() - scrollX) / getClientWidth(viewPager);
                // mPageTransformer is your custom PageTransformer that will do FadeIn/FadeOut animation
                // child is a view that belongs to one of PageFragment or PageSupportFragmnet.
                // transformPos is a position of page relative to the current front-and-center position of the pager. 0 is front and center. 1 is one full page position to the right, and -1 is one page position to the left. 
                mPageTransformer.transformPage(child, transformPos);
            }
        }
    });
}

private int getClientWidth(ViewPager viewPager) {
    return viewPager.getMeasuredWidth() - viewPager.getPaddingLeft() - viewPager.getPaddingRight();
}
```

I haven't tested it, but this should do the trick.

**Comment by ilchenko-peter** [comment]
This feature was implemented in 1.0.4. You can set PageTransformer via `TutorialOptions.Builder#setPageTransformer(ViewPager.PageTransformer pageTransformer)`

**Comment by ghost** [comment]
ok thank you so much for quick replay ...i'll try the solutions.......
